### DISCUSSION TITLE
I'd like to add a Max Width Width feature

### DISCUSSION THREAD

**Discussion by mike-lloyd03** [pop-os/shell#1752] [General]
I'm interested in adding a max window width feature so tiled windows don't take up the whole screen on ultra-wide monitors. This would prevent any tiled window from being wider than the configured max width. So if I have a monitor of 3440 pixels wide, I could configure pop-shell to only create windows that are maximum 2400 pixels wide. If there's a single window on the workspace, it would be centered with open space on the left and right.

I am comfortable with Typescript and familiar with GTK and GJS. However, this is a large project and my time is somewhat limited unfortunately.  

If I were to implement this, where should I start? I took a brief look through `src/tiling.ts` and that seems like the right place to start but any guidance would be greatly appreciated.

Furthermore, with the pop-os team now focusing their engineering efforts on Cosmic, what is the likelihood of this being reviewed and merged?

Thank you.

**Comment by jacobgkau** [comment]
> Furthermore, with the pop-os team now focusing their engineering efforts on Cosmic, what is the likelihood of this being reviewed and merged?

Our engineers are busy with COSMIC Epoch and may not be able to provide coding assistance, but I can speak to this one from a QA perspective. If you put the feature behind a hidden option (e.g. in the config file and/or the extension settings panel), and don't put it in the top-right menu, then we would be likely to review it for regressions to officially supported functionality (and check that the new feature works for at least basic usage) and then merge it if there are no engineering concerns about the code maintainability.

The future roadmap for user-facing features (exposed in the top-right menu) is admittedly a little unclear right now. You would be well within your rights to fork the project in the event you find you want to make substantial changes that we're not comfortable merging into our organization's codebase.

Some alternative tiling-related GNOME Shell extensions to take a look at are [Tiling Shell](https://github.com/domferr/tilingshell) and [Forge](https://github.com/forge-ext/forge) (currently unmaintained). It would also be a great idea to see if the max-width feature you're describing has a place in [COSMIC Comp](https://github.com/pop-os/cosmic-comp), although I understand Rust is different than TypeScript if you're just looking to make a quick contribution and aren't already familiar with Rust.

**Reply by mike-lloyd03** [reply]
Thanks for the response and I'm many weeks late on responding here.

In fact, I very much love Rust and can't wait for Cosmic to be ready for daily use. I'd like to implement this feature now since it impacts my current workflow, but if I can get it working, I'd happily consider making a PR to Cosmic with a similar feature.

Let me dig around in the code a bit and see what I can come up with. Thanks for your willingness to consider merging the feature.

**Reply by mike-lloyd03** [reply]
Okay I took a stab at it in #1762. Let me know what you think.
